ethcore/snapshot/src/watcher.rs

```diff
@@ -86,17 +86,16 @@ impl Watcher {
 	/// Create a new `Watcher` which will trigger a snapshot event
 	/// once every `period` blocks, but only after that block is
 	/// `history` blocks old.
-	pub fn new<F>(client: Arc<Client>, sync_status: F, channel: IoChannel<ClientIoMessage<Client>>, period: u64, history: u64) -> Self
-		where F: 'static + Send + Sync + Fn() -> bool
+	pub fn new<F, C>(client: Arc<dyn BlockInfo>, sync_status: F, channel: IoChannel<ClientIoMessage<C>>, period: u64, history: u64) -> Self
```

Comment:
Why doesn't it use the same `C` from `Mutex<IoChannel<ClientIoMessage<C>>>`, i.e. why introduce a separate generic type in the constructor?

Reply:
I think they are the same? Maybe you misread the diff thinking `new()` was part of the impl of `Broadcast for Mutex<IoChannel<ClientIoMessage<C>>>`? Or I'm misunderstanding! `Watcher` is not parametrized.

Reply:
Ah, right, I misread the diff, sorry :/